scripts/data/client.py

```diff
@@ -125,101 +149,104 @@ def process_batch(job):
         else:
             error_match = re.search(r"error='([^']*)'", error)
             message = error_match.group(1) if error_match else ""
-
         logger.error(f"{job} error: {message}")
         logger.debug(f"Full error while processing: {job}:\n{error}")
     else:
-        match = re.search(r"gas_spent=(\d+)", result.stdout)
+        match = re.search(r"gas_spent=(\d+)", stdout)
         gas_info = f"gas spent: {int(match.group(1))}" if match else "no gas info found"
         logger.info(f"{job} done, {gas_info}")
         if not match:
-            logger.warning(f"{job}: not gas info found")
+            logger.warning(f"{job}: no gas info found")
+
+
+def process_batch(job):
+    if shutdown_event.is_set():
+        logger.info(f"Shutdown detected, skipping job: {job}")
+        return
+
+    arguments_file = job.batch_file.as_posix().replace(".json", "-arguments.json")
+    try:
+        with open(arguments_file, "w") as af:
+            af.write(str(format_args(job.batch_file, False, False)))
+        run_process(arguments_file, job)
+    except Exception as e:
+        logger.error(f"Error processing batch {job}: {e}")
 
 
-# Producer function: Generates data and adds jobs to the queue
 def job_producer(job_gen):
     global current_weight
 
     try:
         for job, weight in job_gen:
-            # Wait until there is enough weight capacity to add the new block
+            if shutdown_event.is_set():
```

Comment:
Is this check necessary? `shutdown_event` is checked few lines below.

Reply:
I've removed redundant `shutdown` checks in `job_generator`